test/test-golden/nested-embed.adoc

```diff
@@ -101,7 +101,7 @@ Dictionary object with extension-specific objects.
 
 * **Type**: <<reference-extension,`extension`>>
 * **Required**: No
-* **Type of each property**: Extension
+* **Type of each property**: `object`
```

Comment:
Are we sure we want this change?  The glTF extensions aren't just simple JSON objects, they do follow the Extension schema, which is what enables nesting.  (For example, the "clearcoat" extension may have a "texture transform" extension nested inside of it).

Reply:
Also it explicitly allows `extras` inside of extensions.

Reply:
Maybe I got confused here at some point, or am overlooking something.

An `extension` object is defined as a plain JSON object, as of the [`extensions.schema.json`](https://github.com/KhronosGroup/glTF/blob/cdf649165a6dd2181936b959aa7e2ab7d0d9c3a8/specification/2.0/schema/extension.schema.json). It is only an `object`, without additional constraints for its `properties`. Its `additionalProperties` are defined to be `object`, too. 

From what you said about 'nesting', it sounds like the _values_ in the extension dictionary should be constrained to be ~"something that can (optionally) contain another 'extensions' property" (and maybe an optional 'extras' property). So the _values_ could be required to be a [`glTFProperty`](https://github.com/KhronosGroup/glTF/blob/cdf649165a6dd2181936b959aa7e2ab7d0d9c3a8/specification/2.0/schema/glTFProperty.schema.json), but this 1. does not seem to be the case, and 2. (from the tip of my head) would not change anything in the meaning, compared to just using `object` - or does it?

Reply:
You might be right.  I've always been under the impression that the `extensions` dictionary object contains a dictionary of named `Extension` type objects, but it's been a while since I investigated this and I'm not sure.

Maybe @lexaknyazev can double check?  This is a far-reaching change because every type of object in a glTF has an `extensions` dictionary on it, and I feel like the type of entry in that dictionary has changed here.  At the very least, it will produce a lot of diffs across the entire section 5 "Properties Reference."

Reply:
On the nitpicking level, some confusion might be caused by the fact that the type of the `extensions` property is defined in the `extension.schema.json`. The schema actually _does_ define the structure of the extension**s** property (and we might consider renaming that file, if this doesn't cause too much trouble - at least, it would match the name of the `extras` (not `extra`) schema file...)